lib/primitives/base-element.js

```diff
@@ -52,113 +59,86 @@ module.exports = function(registry) {
       return _.extend(initial, options || {});
     },
 
-    /*
-     * Converts everything in `meta` into values, unless any top-level element
-     * contains additional metadata or attributes, in which case it gets
-     * refracted.
-     */
-    convertMetaToRefract: function(functionName) {
-      var meta = {};
-      this.meta.keys().forEach(function (key) {
-        var item = this.meta.get(key);
-        if (item.meta.keys().length || item.attributes.keys().length) {
-          // Additional information, so we need to refract this element!
-          meta[key] = item[functionName]();
-        } else if (item.element === 'array') {
-          // This is a plain array, but maybe it contains elements with
-          // additional information? Let's see!
-          var values = [];
-          for (var index = 0; index < item.length; index++) {
-            var subItem = item.get(index);
-            if (subItem.meta.keys().length || subItem.attributes.keys().length) {
-              values.push(subItem[functionName]());
-            } else {
-              values.push(subItem.toValue());
-            }
-          }
-          meta[key] = values;
-        } else {
-          meta[key] = item.toValue();
-        }
-      }, this);
-      return meta;
-    },
+    convertKeyToRefract: function (ret, key, item) {
+      if (this.shouldRefract(item)) {
+        ret[key] = item.toRefract();
+        return;
+      }
 
-    /*
-     * Some attributes may be elements. This is domain-specific knowledge, so
-     * a subclass *MUST* define the attribute element names to convert. This
-     * method handles the actual serialization to refract.
-     */
-    convertAttributesToRefract: function(functionName) {
-      var attributes = {};
-      var keys = this.attributes.keys();
-
-      for (var i = 0; i < keys.length; i++) {
-        var key = keys[i];
-        if (this._attributeElementKeys.indexOf(key) !== -1) {
-          attributes[key] = this.attributes.get(key)[functionName]();
-        } else if (this._attributeElementArrayKeys.indexOf(key) !== -1) {
-          attributes[key] = [];
-          var items = this.attributes.get(key);
-          for (var j = 0; j < items.length; j++) {
-            attributes[key].push(items.get(j)[functionName]());
+      if (item.element === 'array') {
```

Comment:
That will be handled above in the `this.shouldRefract`. If any element is a superclass of `array`, it needs to be refracted since we need to print the element name.